addons/mail/static/src/models/composer_view/composer_view.js

```diff
@@ -153,26 +152,59 @@ function factory(dependencies) {
                 );
             }
             const recordReplacement = this.activeSuggestedRecord.getMentionText();
-            const updateData = {
-                isLastStateChangeProgrammatic: true,
-                textInputContent: textLeft + recordReplacement + ' ' + textRight,
-                textInputCursorEnd: textLeft.length + recordReplacement.length + 1,
-                textInputCursorStart: textLeft.length + recordReplacement.length + 1,
-            };
             // Specific cases for channel and partner mentions: the message with
             // the mention will appear in the target channel, or be notified to
             // the target partner.
             switch (this.activeSuggestedRecord.constructor.modelName) {
                 case 'mail.thread':
-                    Object.assign(updateData, { mentionedChannels: link(this.activeSuggestedRecord) });
+                    this.composer.pendingMessage.update({ mentionedChannels: link(this.activeSuggestedRecord) });
                     break;
                 case 'mail.partner':
-                    Object.assign(updateData, { mentionedPartners: link(this.activeSuggestedRecord) });
+                    this.composer.pendingMessage.update({ mentionedPartners: link(this.activeSuggestedRecord) });
                     break;
             }
+            const updateData = {
+                isLastStateChangeProgrammatic: true,
+                textInputContent: textLeft + recordReplacement + ' ' + textRight,
+                textInputCursorEnd: textLeft.length + recordReplacement.length + 1,
+                textInputCursorStart: textLeft.length + recordReplacement.length + 1,
+            };
             this.composer.update(updateData);
         }
 
+        /**
+         * Remove suggestion from the pending message when user delete the suggestion
```

Comment:
Remove "when user delete the suggestion", because it shouldn't tell where it's called.
It looks more appropriate to make this as a method of `PendingMessage` rather than `ComposerView`.